tensorflow/core/kernels/string_split_op.cc

```diff
@@ -85,7 +98,9 @@ class StringSplitOp : public OpKernel {
     int64 max_num_entries = 0;
     std::vector<int64> num_indices(batch_size);
     for (int64 i = 0; i < batch_size; ++i) {
-      std::vector<string> parts = Split(input_vec(i), delimiter, skip_empty_);
+      std::vector<string> parts;
+      OP_REQUIRES_OK(
+          ctx, Split(input_vec(i), delimiter, skip_empty_, encoding_, &parts));
```

Comment:
If `encoding_ == "utf8"`, should the kernel fail if the delimiter is not valid UTF8?

Reply:
Thanks @asimshankar. The PR has been updated so that delimiter will be checked for properly encoding.